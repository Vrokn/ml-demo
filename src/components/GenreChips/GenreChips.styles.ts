import styled from "@emotion/styled";

export const ChipsContainer = styled.div(() => ({
  display: 'flex',
  alignItems: 'center',
  gap: '0.5em',
  flexWrap: 'wrap',
}));