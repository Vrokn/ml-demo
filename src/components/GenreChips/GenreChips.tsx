import React from "react";
import { ChipsContainer } from "./GenreChips.styles";
import { Chip, Typography } from "@mui/material";
import { Genre } from "../../interfaces/genre";

const GenreChips = ({ genres, label, color }: { genres: Genre[]; label: string; color: 'primary' | 'secondary'; }) => (
  <ChipsContainer>
    <Typography variant="caption">{label}:</Typography>
    {genres.map((genre: Genre) => (
      <Chip key={genre.id} label={genre.name} color={color} />
    ))}
  </ChipsContainer>
);

export default GenreChips;