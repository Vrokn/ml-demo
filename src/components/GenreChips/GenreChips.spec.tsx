import React from 'react';
import { render } from '@testing-library/react';
import GenreChips from './GenreChips';

const mockGenres = [
  { id: 1, name: 'Pop' },
  { id: 2, name: 'Rock' },
];

describe('<GenreChips />', () => {
  it('renders the chips with correct genre names', () => {
    const { getByText } = render(<GenreChips genres={mockGenres} label="Genres" color="primary" />);
    expect(getByText('Pop')).toBeInTheDocument();
    expect(getByText('Rock')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<GenreChips genres={mockGenres} label="Genres" color="primary" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
