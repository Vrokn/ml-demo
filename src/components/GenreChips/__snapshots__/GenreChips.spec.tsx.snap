// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenreChips /> matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-4z29z6"
  >
    <span
      class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
    >
      Genres:
    </span>
    <div
      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-filledPrimary css-uy9ylj-MuiChip-root"
    >
      <span
        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
      >
        Pop
      </span>
    </div>
    <div
      class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-filledPrimary css-uy9ylj-MuiChip-root"
    >
      <span
        class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
      >
        Rock
      </span>
    </div>
  </div>
</DocumentFragment>
`;
